src/types.jl

```diff
@@ -119,26 +114,13 @@ struct SCSInfo
     solveTime::Cdouble
 end
 
-SCSInfo() = SCSInfo(0, convert(Int128, 0), convert(Int128, 0), 0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0)
+SCSInfo() = SCSInfo(0, ntuple(_ -> zero(Cchar), 32), 0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0)
 
 function raw_status(info::SCSInfo)
-    chars = UInt8[]
-    eos = false
-    for status in [info.status1, info.status2]
-        for i in 1:16
-            char = status & 0xff
-            if iszero(char)
-                eos = true
-                break
-            end
-            push!(chars, char)
-            status >>= 8
-        end
-        if eos
-            break
-        end
-    end
-    return String(chars)
+    s = collect(info.status)
+    len = findfirst(iszero, s) - 1
+    # There is no method String(::Vector{Cchar}) so we convert to `UInt8`.
```

Comment:
you could `String(reinterpret(UInt8, s)[1:len])` as well

Reply:
If we use the fact that `UInt8` has the same binary representation of `Cchar`, we might as well store it as `UInt8` in the struct

Reply:
yes, this is why my intial suggestion was to go with `Cuchar`; but this is a matter of naming/taste, so it can stay as it is now 

(but this convert via array comprehension makes me wince :))

Reply:
Yes but I would rather have the code not depend on equivalence of bit representation just to have `raw_status` be slightly faster. What's wrong with array comprehension :-P ? How would you write it?

Reply:
I'm not feeling strongly about it; that bike-shedding was intended as a joke ;-).
I'd probably go for `reinterpret(UInt8, s[1:len])` if we really do care about this one allocation (we don't:).